zowe/zowe-explorer-vscode
# feat: Detect JWT expiration in token values and prompt user if expired

## Proposed changes

Implements JWT expiration checks for profiles using a token for authentication.

- Prompts the user if the JWT could be successfully decoded from the token **and** the expire time/date has elapsed
- Skips LTPA token values (cannot be decoded w/o private key, only given to servers generating the tokens)

## How to test

1. Ensure that dependencies are up-to-date using `pnpm install`
2. Log into a profile using API ML or z/OSMF using JWT token authentication.
3. Wait for the token to expire ⏳ 
4. Once expired, attempt to search again - notice the login prompt will appear instead of the search prompt, as the token has expired

## Release Notes

Milestone: 3.1.0

Changelog:

- Added expired JWT token detection for profiles in each tree view (Data Sets, USS, Jobs). When a user performs a search on a profile, they are prompted to log in if their token expired. [#3175](https://github.com/zowe/zowe-explorer-vscode/issues/3175)

## Types of changes




## Checklist



**General**

  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**


**Deployment**

  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
